DB-11791 table scan benchmark with physical splits

hatyo opened the pull request:
<!--- Provide a general summary of your changes in the Title above. -->

## Short Description
Introduces a new micro-benchmark for base table scan against varying number of physical splits.

## Long Description
As part of studying and improving the optimiser cost model, we introduce in this PR add a new micro-benchmark for base table scans with varying number of physical to gain a better understanding on how scans perform with different number of splits (specially in OLAP).

## How to test
Run the tests in `ScanBenchmark`, preferably against a cluster running in cloud environment, similar to what's explained [here](https://github.com/splicemachine/spliceengine/pull/5360#issue-605614457), basically:

1. Prepare the necessary jar files as a tarball.
```
$ mvn -Pcore,cdh6.3.0 clean install -DskipTests
$ mkdir benchmark
$ cp ./platform_it/src/main/resources/info-log4j.properties /Users/<your username>/.m2/repository/junit/junit/4.12/junit-4.12.jar /Users/<your username>/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar ./splice_machine/target/splice_machine-<VERSION>-tests.jar ./platform_it/target/dependency/log4j-1.2.17.jar ./platform_it/target/dependency/db-client-<VERSION>.jar benchmark
$ tar czvf benchmark.tar.gz ./benchmark
$ 
```
2. Upload the tarball to your cluster. You need to be authorized to the k8s environment before doing this step.
```
$ kubectl cp benchmark.tar.gz splicedb-hregion-0:/home/hbase/ -n <your cluster namespace>
```
3. Log into the `splicedb-hregion-0` pod of your cluster:
```
$ kubectl exec -it splicedb-hregion-0 -n <your cluster namespace> -- bash
```
4. Once in pod, run the benchmark:
```
$ cd home/hbase
$ tar zxvf benchmark.tar.gz
$ java -Dlog4j.configuration=file:./benchmark/info-log4j.properties -cp 'benchmark/*' org.junit.runner.JUnitCore com.splicemachine.benchmark.<benchmark name>
```

martinrupp commented on splice_machine/src/test/java/com/splicemachine/benchmark/ScanBenchmark.java:
```diff
@@ -0,0 +1,699 @@
+        curSize.set(0);
+        runBenchmark(NUM_CONNECTIONS, () -> populateTables(tableSize));
+
+        testStatement.execute(String.format("call syscs_util.syscs_flush_table('%s', '%s')", SCHEMA, BASE_TABLE));
+        LOG.info("Collect statistics");
+        try (ResultSet rs = testStatement.executeQuery("ANALYZE SCHEMA " + SCHEMA)) {
```
methodWatcher.execute
methodWatcher.executeQuery

arnaud-splice reviewed (commented):
+1 once Martin's comment have been addressed.

hatyo commented:
@martinrupp, as discussed, I created a ticket for addressing your comments, since your comments apply not only to this microbechmark but to all other microbenchmarks such as:
- FeatureStoreBenchmark
- IndexLookupBenchmark
- JoinBenchmark
- PKLookupBenchmark
I created ticket DB-12096 to refactor all these microbenchmarks accordingly.

carolp-503 reviewed (approved):
+1
